src/jdk.incubator.jextract/share/classes/jdk/incubator/jextract/tool/JavaSourceBuilder.java

```diff
@@ -125,24 +131,32 @@ protected void addLayout(String elementName, MemoryLayout layout) {
 
     private void addLayout(MemoryLayout l) {
         if (l instanceof ValueLayout) {
-            boolean found = false;
-            for (Field fs : MemoryLayouts.SysV.class.getDeclaredFields()) {
-                try {
-                    MemoryLayout constant = (MemoryLayout)fs.get(null);
-                    if (l.name().isPresent()) {
-                        constant = constant.withName(l.name().get());
-                    }
-                    if (constant.equals(l)) {
-                        found = true;
-                        sb.append("MemoryLayouts.SysV." + fs.getName());
-                        break;
-                    }
-                } catch (ReflectiveOperationException ex) {
-                    throw new AssertionError(ex);
+            SystemABI.Type type = l.abiType().orElseThrow(()->new AssertionError("Should not get here: " + l));
+            if (type == SystemABI.Type.LONG_DOUBLE) {
+                if (ABI != SystemABI.ABI_SYSV) {
+                    throw new RuntimeException("long double is supported only for SysV ABI");
+                } else {
+                    sb.append("C_LONGDOUBLE");
                 }
-            }
-            if (!found) {
-                throw new AssertionError("Should not get here: " + l);
+            } else {
```

Comment:
`long double` is a standard type, so it should be fine to support on other platforms as well. e.g. Windows: https://docs.microsoft.com/en-us/cpp/cpp/fundamental-types-cpp?view=vs-2019

Looks like the constant is just missing from MemoryLayouts.WinABI. Should be 8 bytes there.
```java
public static final ValueLayout C_LONGDOUBLE = MemoryLayout.ofValueBits(64, ByteOrder.LITTLE_ENDIAN)
        .withAnnotation(AbstractLayout.NATIVE_TYPE, SystemABI.Type.LONG_DOUBLE);
```

Looks like on arm it is 16 bytes, so you can copy the impl from SysV.

Reply:
Nice suggestion. Will add long double for other ABIs